common/src/unifyfs_meta.c

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2020, Lawrence Livermore National Security, LLC.
+ * Produced at the Lawrence Livermore National Laboratory.
+ *
+ * Copyright 2020, UT-Battelle, LLC.
+ *
+ * LLNL-CODE-741539
+ * All rights reserved.
+ *
+ * This is the license for UnifyFS.
+ * For details, see https://github.com/LLNL/UnifyFS.
+ * Please read https://github.com/LLNL/UnifyFS/LICENSE for full license text.
+ */
+
+#include <string.h>
+#include <openssl/md5.h>
+
+#include "unifyfs_meta.h"
+
+/**
+ * Hash a file path to a 64-bit unsigned integer using MD5
+ * @param path absolute file path
+ * @return hash value
+ */
+uint64_t compute_path_md5(const char* path)
+{
+    unsigned long len;
+    unsigned char digested[16] = {0};
+
+    len = strlen(path);
+    MD5((const unsigned char*) path, len, digested);
+
+    /* construct uint64_t hash from first 8 digest bytes */
+    uint64_t hash = 0;
+    for (int i = 0; i < 8; i++) {
+        int shift_bits = 8 * (7 - i);
+        uint64_t byteval = (uint64_t)(digested[i]);
+        hash |= (byteval << shift_bits);
+    }
```

Comment:
```C
#include <endian.h>
uint64_t hash = be64toh( * ((uint64_t *) digested));
```

Reply:
@tonyhutter Thanks for the reference - hadn't encountered those conversion functions before. Can we safely assume endian.h is available on our target platforms, or should I add a configure check? I found it was added back in glibc 2.9, which seems pretty old, so I'm guessing it's safe to not check.

Reply:
It's safe to skip the check - glibc 2.9 is 2008-2009 era.

Reply:
done